src/core/utils/fileutils.cpp

```diff
@@ -318,3 +321,250 @@ void FileUtils::addImageStamp( const QString &imagePath, const QString &text )
+  }
+
+  if ( !directory.exists() )
+  {
+    if ( !directory.mkpath( "." ) )
+    {
+      qDebug() << QStringLiteral( "Failed to create directory for file: %1. This may be due to permission restrictions." ).arg( filePath );
+      return false;
+    }
+  }
+
+  if ( file.open( QIODevice::WriteOnly ) )
+  {
+    qint64 bytesWritten = file.write( content );
+    file.close();
+
+    if ( bytesWritten != content.size() )
+    {
+      qDebug() << QStringLiteral( "Failed to write all data to file: %1" ).arg( filePath );
+      return false;
+    }
+
+    return true;
+  }
+  else
+  {
+    QString errorMsg = file.errorString();
+    if ( file.error() == QFile::PermissionsError )
+    {
+      errorMsg += QStringLiteral( " - This may be due to platform security restrictions." );
+    }
+    qDebug() << QStringLiteral( "Failed to open file for writing: %1" ).arg( errorMsg );
+    return false;
+  }
+}
+
+QVariantMap FileUtils::getFileInfo( const QString &filePath )
+{
+  QVariantMap info;
+
+  if ( !isWithinProjectDirectory( filePath ) )
+  {
+    qWarning() << QStringLiteral( "Security warning: Attempted to access file info outside project directory: %1" ).arg( filePath );
+    info["exists"] = false;
+    info["error"] = QStringLiteral( "Access denied: File is outside the current project directory" );
+    return info;
+  }
+
+  QFileInfo fileInfo( filePath );
+  QFile file( filePath );
+
+  if ( fileInfo.exists() )
+  {
+    info["exists"] = true;
+    info["fileName"] = fileInfo.fileName();
+    info["filePath"] = fileInfo.absoluteFilePath();
+    info["fileSize"] = fileInfo.size();
+    info["lastModified"] = fileInfo.lastModified();
+    info["suffix"] = fileInfo.suffix();
+
+    QMimeDatabase db;
+    info["mimeType"] = db.mimeTypeForFile( filePath ).name();
+
+    QByteArray md5Hash = fileChecksum( filePath, QCryptographicHash::Md5 );
+    if ( !md5Hash.isEmpty() )
+    {
+      info["md5"] = md5Hash.toHex();
+    }
+    else
+    {
+      info["md5Error"] = QStringLiteral( "Could not calculate MD5 hash - possibly due to permission restrictions" );
+    }
+
+    if ( file.open( QIODevice::ReadOnly ) )
+    {
+      file.close();
+
+      QByteArray content = readFileContent( filePath );
+
+      if ( !content.isEmpty() || fileInfo.size() == 0 )
+      {
+        info["content"] = content;
+        info["readable"] = true;
+      }
+      else
+      {
+        info["readable"] = false;
+        info["readError"] = QStringLiteral( "Failed to read file content" );
+      }
+    }
+    else
+    {
+      info["readable"] = false;
+      info["readError"] = file.errorString();
+
+      if ( file.error() == QFile::PermissionsError )
+      {
+        info["readError"] = info["readError"].toString() + QStringLiteral( " - This may be due to platform security restrictions" );
+      }
```

Comment:
Could we add a bool fetchContent parameter (false by default) to this function? I feel like there are quite a few scenarios when we won't want to get the content at all.

Reply:
Very good! Will add it!